tools/generators/files_and_groups/src/Generator/Elements/Elements+RootElements.swift

```diff
@@ -0,0 +1,80 @@
+import PBXProj
+
+extension Generator {
+    enum Elements {}
+}
+
+extension Generator.Elements {
+    static func rootElements(
+        pathTree: PathTreeNode,
+        workspace: String,
+        elementAttributes: Generator.Environment.Elements.AttributesFunction
+    ) -> (
+        elements: [Element],
+        pathsToIdentifiers: [BazelPath: String],
+        knownRegions: Set<String>,
+        resolvedRepositories: [ResolvedRepository]
+    ) {
+        let resolvedRepositories: [ResolvedRepository] = [
+            .init(sourcePath: ".", mappedPath: workspace),
+        ]
+
+        // FIXME: Implement
+
+        var elements: [Element] = []
+        for node in pathTree.children {
+            switch node.name {
+            case "external":
+                elements.append(
+                    Generator.Elements.specialRootGroup(
+                        specialRootGroupType: .legacyBazelExternal,
+                        childIdentifiers: []
+                    )
+                )
+
+            case "bazel-out":
+                elements.append(
+                    Generator.Elements.specialRootGroup(
+                        specialRootGroupType: .bazelGenerated,
+                        childIdentifiers: []
+                    )
+                )
+
+            default:
+                break
+            }
+        }
+
+        // FIXME: Should this sorting happen here, or after the call?
```

Comment:
You mean in the call site vs here in the impl? Is `sortOrder` ever "conditional" or should be always respected? If so should probably live here but I don't know if you have other scenarios/edge cases in mind?

Reply:
I mean after this function returns the values, should the caller sort it. It's not conditional, but my thought was more to limit the responsibility of each thing.

I agree with you though, and since I wrote the FIXME the "purpose" of this function has been tuned in, and it should sort them.